miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/RealMiniAppDisplay.kt

```diff
@@ -1,97 +1,37 @@
 package com.rakuten.tech.mobile.miniapp.display
 
 import android.annotation.SuppressLint
+import android.app.Activity
 import android.content.Context
 import android.view.View
-import android.view.ViewGroup
-import android.webkit.WebView
-import android.widget.FrameLayout
-import androidx.annotation.VisibleForTesting
+import androidx.core.app.ActivityCompat
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.OnLifecycleEvent
-import androidx.webkit.WebViewAssetLoader
 import com.rakuten.tech.mobile.miniapp.MiniAppDisplay
 import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridge
-import java.io.File
-
-private const val SUB_DOMAIN_PATH = "miniapp"
-private const val MINI_APP_INTERFACE = "MiniAppAndroid"
 
 @SuppressLint("SetJavaScriptEnabled")
 internal class RealMiniAppDisplay(
-    context: Context,
     val basePath: String,
     val appId: String,
-    miniAppMessageBridge: MiniAppMessageBridge
-) : MiniAppDisplay, WebView(context), WebViewListener {
-
-    private val miniAppDomain = "mscheme.$appId"
-    private val customScheme = "$miniAppDomain://"
-    private val customDomain = "https://$miniAppDomain/"
+    val miniAppMessageBridge: MiniAppMessageBridge
+) : MiniAppDisplay {
 
-    init {
-        layoutParams = FrameLayout.LayoutParams(
-            ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT
-        )
+    private var miniAppWebView: MiniAppWebView? = null
 
-        settings.javaScriptEnabled = true
-        addJavascriptInterface(miniAppMessageBridge, MINI_APP_INTERFACE)
-        miniAppMessageBridge.setWebViewListener(this)
-
-        settings.allowUniversalAccessFromFileURLs = true
-        settings.domStorageEnabled = true
-        settings.databaseEnabled = true
-        webViewClient = MiniAppWebViewClient(context, getWebViewAssetLoader(), customDomain, customScheme)
-
-        loadUrl(getLoadUrl())
-    }
-
-    override fun getMiniAppView(): View = this
+    override fun getMiniAppView(activityContext: Context): View? =
+        if (activityContext is Activity || activityContext is ActivityCompat) {
```

Comment:
I guess we should let the hostapps provide any kind of `context` they want. In documentation, we recommend using `ActivityContext` with the reasonable explaination. 

Furthermore, using `ActivityContext` causes crash on Android 5 (maybe even 6) so you need to update the `androidx.appcompat` like [this](https://github.com/rakutentech/android-miniapp/pull/89/commits/7602ff5965a6087b3938f6c8e8a487f974d39aaa).